capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -401,17 +402,33 @@ else if (key.equals("remove_"+name)) {
   }
 
   /**
-   * This method executes the given post request, creating the new object and returning it with its ID appended.
+   * This method converts the request into a json, and feeds it to the post helper
    * @param {collectionReference} reference to the appropriate database collection
    * @param {request} the request sent to the backend
+   * @param {response} the response returned from the backend
    * @param {genericClass} a Generic class, to be used in casting objects retrieved from the database
    * @param {requiredFields} a list of the names of fields deemed "necessary" for creation of an object.
    * @return T the object created by the post request, with an ID field generated by firebase
    */
+
   public static <T extends BaseEntity> T post(CollectionReference collectionReference, HttpServletRequest request, 
       HttpServletResponse response, GenericClass<T> genericClass, List<String> requiredFields) throws IOException {
-    Map<String, Object> constructorFields = new HashMap<>();
     JsonObject jsonObject = Utility.createRequestBodyJson(request);
+    return postHelper(collectionReference, jsonObject, response, genericClass, requiredFields);
+  }
+
+  /**
+   * This method executes the given post request, creating the new object and returning it with its ID appended.
+   * @param {collectionReference} reference to the appropriate database collection
+   * @param {jsonObject} the request body as a json
+   * @param {response} the response returned from the backend
+   * @param {genericClass} a Generic class, to be used in casting objects retrieved from the database
+   * @param {requiredFields} a list of the names of fields deemed "necessary" for creation of an object.
+   * @return T the object created by the post request, with an ID field generated by firebase
+   */
+  public static <T extends BaseEntity> T postHelper(CollectionReference collectionReference, JsonObject jsonObject, 
```

Comment:
This is a good fix! I'm glad we found a way to make sure we only read the request body once!

Reply:
Thanks!